lcm/src/lcm/mod.rs

```diff
@@ -0,0 +1,168 @@
+use std::env;
+use std::collections::HashMap;
+use std::time::Duration;
+use regex::Regex;
+
+mod providers;
+use self::providers::udpm::UdpmProvider;
+
+use Message;
+use error::*;
+
+/// Convenience macro for dispatching functions among providers.
+macro_rules! provider
```

Comment:
I really like this macro.

Reply:
I was really pleased to find that it was a valid macro.